docs/notebooks/utils/working_with_catalogs.ipynb

```diff
@@ -4,17 +4,96 @@
     "[Obspy's event representation](https://docs.obspy.org/packages/obspy.core.html#event-metadata) is based on the [FDSN](http://www.fdsn.org/) [QuakeML standard](https://quake.ethz.ch/quakeml/), which is very comprehensive, and arguably the best standard available. However, It can be a bit difficult to work with the `Catalog` object (and friends) for a few reasons:\n",
     "\n",
     "    1. Often the desired data is deeply nested and hard to aggregate\n",
     "    \n",
     "    2. Identifying data relations depends on the complex behavior of Obspy's `ResourceIdentifier`\n",
     "    \n",
-    "    3. Preferred objects are often not set\n",
+    "    3. Preferred objects (eg origin, magnitude, etc.) are often not set\n",
     "    \n",
-    "ObsPlus tries to solve all of these problems. The first is addressed by the [DataFrame Extractor](../utils/dataframeextractor.ipynb). The second and third are addressed by a collection of catalog validators that attempt to ensure all resource_ids point to the correct objects, preferred objects are set, as well as preform other sanity checks. The default event validation function in ObsPlus is a bit opinionated and was built specifically for the NIOSH flavor of seismic event, but you may still find it useful. Additionally, you can create your own validation namespace and define validators for your own data/schema as described by the [validators documentation](validators.ipynb)."
+    "ObsPlus tries to solve all of these problems. The first is addressed by the [DataFrame Extractor](../utils/dataframeextractor.ipynb) and other tree transversal tools. The second and third are addressed by a collection of catalog validators "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Catalog Navigation\n",
+    "\n",
+    "If you only need to extract information contained in a catalog there are various examples of creating dataframes from different parts of the catalog in the [events_to_pandas section](../datastructures/events_to_pandas.ipynb). \n",
+    "\n",
+    "If, however, you need to maintain the tree structure the `yield_object_parent_attr` function can be very useful. For example, let's assume we want to sabotage a seismic analyst by adding noise to his/her pick times. We could do that like so: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# \n",
+    "import numpy as np\n",
+    "import obspy\n",
+    "import obspy.core.event as ev\n",
+    "import obsplus\n",
+    "from obsplus.utils import yield_obj_parent_attr\n",
+    "\n",
+    "cat = obsplus.load_dataset('crandall_test').event_client.get_events()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# iterate picks, add noise to pick time (-1 to 1 seconds, uniform dist.)\n",
+    "for pick, _, _ in yield_obj_parent_attr(cat, cls=ev.Pick):\n",
```

Comment:
Because this is the first example, it might be useful to use `pick, parent, attr` for clarity as to what the `yield_obj_parent_attr` yields (although the name is ideal!).